### ISSUE TITLE
Combining Kopf with Nginx

### ISSUE THREAD

**Issue by JaminVP** [lmenezes/elasticsearch-kopf#334]
Greetings, this is a followup on #229.

I'm getting the error message mentioned in #229 when connecting to Kopf through HTTPS. Previously my Elasticsearch cluster was sitting behind Nginx which itself sits next to my company's Apache Server (of which I have little control of) and I was controlling access with the remote IP-address of the entity requesting access.

This worked fine but it was clumsy as admin users didn't have static IP-addresses unlike servers. I'm now trying to keep the static IP-filtering for servers but allow admin access through the Apache HTTPS Server acting as a reverse proxy for Nginx which adds a secure header I can check credentials on. This is how I secured Kibana access. 

I get the Kopf html and css content but I get an error when trying to connect to Elasticsearch as Kopf tries to connect to port 443 on the Apache Server, which I have no control over. The other issue was closed without a clear solution and now I seem to be stuck. Any help?

TLDR: Kopf can't reach Elasticsearch when being reverse proxied through HTTPS

The exact error:
{
  "description": "Unable to get property 'number' of undefined or null reference",
  "number": -2146823281,
  "stack": "TypeError: Unable to get property 'number' of undefined or null reference\n   at Anonymous function (https://company/Kopf/dist/kopf.js:3800:13)\n   at Anonymous function (https://company/Kopf/dist/kopf.js:4306:15)\n   at Anonymous function (https://company/Kopf/dist/lib.js:86:387)\n   at Anonymous function (https://company/Kopf/dist/lib.js:118:14)\n   at l.prototype.$eval (https://company/Kopf/dist/lib.js:131:298)\n   at l.prototype.$digest (https://company/Kopf/dist/lib.js:128:392)\n   at l.prototype.$apply (https://company/Kopf/dist/lib.js:132:62)\n   at l (https://company/Kopf/dist/lib.js:87:173)\n   at S (https://company/Kopf/dist/lib.js:91:303)\n   at D.onload (https://company/Kopf/dist/lib.js:92:321)"
}

**Comment by lmenezes** [comment]
@JaminVP could you check https://github.com/lmenezes/elasticsearch-kopf/issues/330 ?
In short, if your ES is located somewhere different than the server serving kopf, you need two things, as listed here: https://github.com/lmenezes/elasticsearch-kopf#kopf-behind-a-reverse-proxy

Basically, set the kopf path to something like /es, and setting a redirect rule on ngix from /es to the correct place.

Hope this helps :)

**Comment by JaminVP** [comment]
It results in the same error but with the es added to it unfortunately:
Error connecting to [https://company.com:443/es]

"<html><head><title>Apache Tomcat/7.0.40 - Error report</title><style></style> </head><body><h1>HTTP Status 404 - /company/es/</h1><HR size=\"1\" noshade=\"noshade\"><p><b>type</b> Status report</p><p><b>message</b> <u>/company/es/</u></p><p><b>description</b> <u>The requested resource is not available.</u></p><HR size=\"1\" noshade=\"noshade\"><h3>Apache Tomcat/7.0.40</h3></body></html>"

Interesting enough, I can connect to http://localhost:8080/es on the server hosting ES, Kopf and Nginx and it gives me the expected ElasticSearch default JSON-reply.

**Comment by lmenezes** [comment]
but should it be /company/es? and not only /es?

**Comment by JaminVP** [comment]
I'll be a little more specific with the "company".

This is the link I'm trying to access Kopf with: 
https://comp1.comp2.comp3.com/CompanyTools/Kopf/

The Apache HTTPS Reverse proxy has the following rewrite rule:

RewriteRule ^/CompanyTools/Kopf/(.*)$ http://localhost:9201/_plugin/kopf/$1 [P]

A similiar rewrite rule works perfectly for Kibana:
RewriteRule ^/RPTools/Kibana/(.*)$ http://localhost:5600/$1 [P] 

localhost:5600 and localhost:9201 are both assigned to Nginx acting as a second reverse proxy (I barely have access rights for the Apache reverse proxy).
This Nginx reverse proxy can only be accessed on the localhost and will check a header for credentials. The server can only be externally accessed through the Apache HTTPS reverse proxy.
Nginx conf:

    http {

    upstream elasticsearch {
    	server 127.0.0.1:9200;
    }

    server {
	listen 9201;
	
	location ~ /(?<section>.*) {
		set $accesselastic 0;

		#check credentials
		if ($accesselastic = 0){
			return 403;
		}
		
		proxy_pass http://elasticsearch/$section;

	}

    }


    server {

		listen 127.0.0.1:5600;

	
	location ~ /(?<section>.*) {
		set $access 0;
		#check credentials		
		if ($access = 0){
			return 403;
		}

		proxy_pass http://127.0.0.1:5601/$section; 
	}

	
    }

    server {
  	listen       8080;
  	server_name  localhost;

	location ~ ^/es.*$ {
    		proxy_pass http://localhost:9200;
    		rewrite ^/es(.*) /$1 break;
  	}

  	location ~ ^/kopf/.*$ {
    		proxy_pass http://localhost:9200;
    		rewrite ^/kopf/(.*) /_plugin/kopf/$1 break;
  	}
    }

    }


The last server configuration is your suggested configuration for fixing the error. Kopf is installed on the same server as Elasticsearch and Nginx and has the following kopf_external_settings.json file:

     {
     "elasticsearch_root_path": "/es",
     "with_credentials": false,
     "theme": "dark",
     "refresh_rate": 5000
     }
With this setup, I can connect to the Kopf HTML and CSS-files but Kopf can't connect to Elasticsearch.
I hope you can help me fix this with the given configuration files. Currently I'm rather clueless. I just can't understand why I can access Elasticsearch but Kopf can't connect to it, it seems like a configuration error in Kopf but I don't know where to find it.
